src/grid.py

```diff
@@ -0,0 +1,111 @@
+from __future__ import annotations
+from dataclasses import dataclass
+from tabulate import tabulate
+from typing import List, Set, Dict
+from uuid import UUID, uuid4
+
+
+@dataclass(eq=True, frozen=True)
+class Location:
+    x: int
+    y: int
+
+
+class ItemNotFoundException(Exception):
+    pass
+
+
+class Grid:
+    def __init__(self, size_x: int, size_y: int):
+        self.__cells: List[List[GridItem | None]] = [[None for _ in range(size_y)] for _ in range(size_x)]
+        self.__items_index: Dict[UUID, Location] = dict()
+
+    def place(self, map_item: GridItem, loc: Location):
+        self.__items_index[map_item.uuid] = loc
+        self.__cells[loc.y][loc.x] = map_item
+
+    def clear(self, loc: Location) -> GridItem | None:
+        content = self.__cells[loc.y][loc.x]
+        self.__items_index.pop(content.uuid)
+        self.__cells[loc.y][loc.x] = None
+        return content
+
+    def move(self, source: Location, dest: Location):
+        item = self.clear(source)
+        self.place(item, dest)
+
+    def find(self, item) -> Location:
+        try:
+            return self.__items_index.get(item.uuid)
+        except IndexError:
+            raise ItemNotFoundException("Item not in grid")
+
+    def get_adjacent_items(self, loc: Location) -> Set[GridItem]:
+        all_adjacent_indices = {
+                Location(loc.x + 1, loc.y),
+                Location(loc.x + 1, loc.y + 1),
+                Location(loc.x, loc.y + 1),
+                Location(loc.x - 1, loc.y + 1),
+                Location(loc.x - 1, loc.y),
+                Location(loc.x - 1, loc.y - 1),
+                Location(loc.x, loc.y - 1),
+                Location(loc.x + 1, loc.y - 1)
+            }
```

Comment:
This is very complicated 👀 
```suggestion
        all_adjacent_indices = {}
        for x in range(loc.x-1,loc.x+2):
            for y in range(loc.y-1,loc.y+2):
                if x < 0 or y < 0 or x > len(self.__cells)-1 or y > len(self.__cells[0])-1:
                    continue
                all_adjacent_indices.add(Location(x,y))
```
Also, i'd make the if condition a function named `is_out_of_bounds` for readbility